relay-server/src/envelope.rs

```diff
@@ -517,6 +517,14 @@ pub struct ItemHeaders {
     #[serde(default, skip_serializing_if = "is_false")]
     metrics_extracted: bool,
 
+    /// `false` if the sampling decision is "drop".
+    ///
+    /// In the most common use case, the item is dropped when the sampling decision is "drop".
+    /// For profiles with the feature enabled, however, we keep all profile items and mark the ones
+    /// for which the transaction was dropped as `sampled: false`.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```

Comment:
`default` is implied for `Option`, but do we actually need `Option<bool>`, can't we use `bool` with a default of `true` and `skip_serializing_if = "if_true"` almost like we have for `metrics_extracted` above.

Reply:
To me it feels wrong to have a flag that defaults to `sampled: true` even before the sampling decision has been made. If the meaning was flipped, i.e. the flag was called `should_be_dropped`, I would agree.

Reply:
Fair enough, I don't have a strong opinion here. If you want to keep the `Option` you can remove the `default` from the serde annotation